crates/orchestrator/src/routes/job_routes.rs

```diff
@@ -0,0 +1,75 @@
+use std::sync::Arc;
+
+use axum::extract::{Query, State};
+use axum::http::StatusCode;
+use axum::response::{IntoResponse, Response};
+use axum::routing::get;
+use axum::Router;
+use serde::Deserialize;
+use uuid::Uuid;
+
+use crate::config::Config;
+use crate::jobs::{process_job, verify_job, JobError};
+
+#[derive(Deserialize)]
+struct JobParams {
+    id: String,
+}
+
+struct JobResult(Result<(), JobError>);
+
+impl IntoResponse for JobResult {
+    fn into_response(self) -> Response {
+        match self.0 {
+            Ok(_) => (StatusCode::OK, "Job processing completed.").into_response(),
+            Err(e) => (StatusCode::INTERNAL_SERVER_ERROR, format!("Job processing failed: {:?}", e)).into_response(),
+        }
+    }
+}
+
+async fn handle_process_job_request(
+    Query(params): Query<JobParams>,
+    State(config): State<Arc<Config>>,
+) -> impl IntoResponse {
+    let id = match Uuid::parse_str(&params.id) {
+        Ok(id) => id,
+        Err(_) => return (StatusCode::BAD_REQUEST, "Invalid UUID").into_response(),
+    };
+
+    JobResult(process_job(id, config).await).into_response()
+}
+
+async fn handle_verify_job_request(
+    Query(params): Query<JobParams>,
+    State(config): State<Arc<Config>>,
+) -> impl IntoResponse {
+    let id = match Uuid::parse_str(&params.id) {
+        Ok(id) => id,
+        Err(_) => return (StatusCode::BAD_REQUEST, "Invalid UUID").into_response(),
+    };
+
+    JobResult(verify_job(id, config).await).into_response()
+}
+
+pub fn job_routes(config: Arc<Config>) -> Router {
+    Router::new()
+        .nest(
+            "/trigger",
+            Router::new()
+                .route(
+                    "/process-job",
+                    get({
+                        let shared_state = config.clone();
+                        move |path| handle_process_job_request(path, State(shared_state))
+                    }),
+                )
+                .route(
+                    "/verify-job",
+                    get({
+                        let shared_state = config.clone();
+                        move |path| handle_verify_job_request(path, State(shared_state))
+                    }),
+                ),
+        )
+        .with_state(config)
+}
```

Comment:
Something like this looks cleaner right, wdyt? maybre a lil tricky to read tho
```suggestion
pub fn job_routes(config: Arc<Config>) -> Router {
    Router::new()
        .nest("/trigger", trigger_routes(config.clone()))
        .with_state(config)
}

fn trigger_routes(config: Arc<Config>) -> Router {
    Router::new()
        .route("/process-job", get(handle_process_job_request))
        .route("/verify-job", get(handle_verify_job_request))
        .with_state(config)
}
```

Reply:
I think your approach is cleaner, thanks for the suggestion.